src/vaillant/08.hmu.HW5103.tsp

```diff
@@ -0,0 +1,454 @@
+  model PowerConsumptionHmu {
+    @unit("W")
+    value: EXP;
+  }
+
+  // B51A FF32xx
+  /** default *r */
+  @base(MF, 0x1a, 0x5, 0xff, 0x32)
+  model r_3 {
+    @maxLength(3)
+    value: IGN;
+  }
+
+  @inherit(r_3)
+  @ext(0)
+  model YieldHeatingDay {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(1)
+  model YieldCoolingDay {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x2)
+  model YieldHwcDay {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0xe)
+  model YieldHeatingMonth {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0xf)
+  model CopHeatingMonth {
+    @divisor(10)
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x10)
+  model YieldHeating {
+    @unit("kWh")
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x11)
+  model CopHeating {
+    @divisor(10)
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x12)
+  model YieldHwcMonth {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x13)
+  model CopHwcMonth {
+    @divisor(10)
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x16)
+  model YieldHwc {
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x17)
+  model CopHwc {
+    @divisor(10)
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x1e)
+  model CompressorBlocktime {
+    @unit("min")
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x1f)
+  model SupplyTempWeighted {
+    @unit("°C")
+    @divisor(10)
+    value: D2C;
+  }
+
+  @inherit(r_3)
+  @ext(0x20)
+  model FlowTemp {
+    @unit("°C")
+    value: D2C;
+  }
+
+  @inherit(r_3)
+  @ext(0x21)
+  model EnergyIntegral {
+    value: integral;
+  }
+
+/**
+  @inherit(r_3)
+  @ext(0x22)
+  model SourceTempInput {
+    @unit("°C")
+    value: D2C;
+  }
+*/
+
+  @inherit(r_3)
+  @ext(0x23)
+  model CurrentYieldPower {
+    @unit("kW")
+    @divisor(10)
+    value: D1B;
+  }
+
+  @inherit(r_3)
+  @ext(0x24)
+  model CurrentConsumedPower {
+    @unit("kW")
+    @divisor(10)
+    value: D1B;
+  }
+
+  @inherit(r_3)
+  @ext(0x25)
+  model CompressorModulation {
+    @unit("%")
+    value: D2C;
+  }
+
+  @inherit(r_3)
+  @ext(0x2A)
+  model YieldCoolingMonth {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x2B)
+  model CopCoolingMonth {
+    @divisor(10)
+    value: UIN;
+  }
+
+  @inherit(r_3)
+  @ext(0x2C)
+  model YieldCooling {
+    @unit("kWh")
+    @divisor(10)
+    value: energy;
+  }
+
+  @inherit(r_3)
+  @ext(0x2D)
+  model CopCooling {
+    @divisor(10)
+    value: UIN;
+  }
+
+
+  @inherit(r_3)
+  @ext(0x3c)
+  model WaterThroughput {
+    @unit("l/h")
+    value: UIN;
+  }
+
+  @condition(Id.Id.sw, ">=902")
+  @inherit(r_3)
+  @ext(0x3d)
+  model FlowPressure {
+    @unit("bar")
+    @divisor(4)
+    value: D2C;
+  }
+
+  @inherit(r_3)
+  @ext(0x4d)
+  model TotalEnergyUsage {
+    @unit("kWh")
+    value: energy;
+  }
+}
```

Comment:
overall this seems to be a reduced version of the main hmu.tsp, so it might make more sense to add conditions to that. or maybe share the common part in another file instead of duplicating and having to maintain both